coral/src/app/features/connectors/details/components/ConnectorPromotionBanner.tsx

```diff
@@ -0,0 +1,59 @@
+import { PromotionBanner } from "src/app/features/topics/details/components/PromotionBanner";
+import { InternalLinkButton } from "src/app/components/InternalLinkButton";
+
+//@TODO replace when api is ready
+// eslint-disable-next-line import/exports-last
+export type TemporaryConnectorPromotionDetails = {
+  status:
+    | "SUCCESS"
+    | "NOT_AUTHORIZED"
+    | "REQUEST_OPEN"
+    | "NO_PROMOTION"
+    | "FAILURE";
+  sourceEnv?: string;
+  targetEnv?: string;
+  targetEnvId?: string;
+  error?: string;
+  connectorName?: string;
+  sourceConnectorConfig?: string;
+};
+interface ConnectorPromotionBannerProps {
+  connectorPromotionDetails: TemporaryConnectorPromotionDetails;
+  /** In case there is an open request for this connector
+   * it can't be promoted, and we want to show that
+   * information to the user.
+   */
+  hasOpenConnectorRequest: boolean;
+  hasOpenClaimRequest: boolean;
+  connectorName: string;
+}
+
+const ConnectorPromotionBanner = ({
```

Comment:
I'm still kind of unconvinced of the usefulness of such shallow wrappers around reusable components. What are the advantages over using `PromotionBanner` directly inline in `ConnectorOverview` ?

It's not a blocking issue, just a theoretical question I am interested in ^^

I'm guessing ease of testing? But if we already tested `PromotionBanner`, is it necessary to test the specific `ConnectorPromotionBanner`?

Reply:
It's ease of testing but also in a sense of reducing complexity of the higher components. In my experience it's good to have small (composed) components that take care (and are well tested) of one specific thing vs having a really big component that then handles a whole bunch of different responsibilities. 

In this case, the component does not do a whole lot, but it's still more comfortable testing the different cases in separation with it. It also has a bit clearer of an API then `PromotionBanner` which makes it a bit more speaking when used (imho).

Reply:
I agree with what you are saying, I just am not sure that creating a shallow wrapper component like this is helping achieving those goals. As I see it, adding a level of indirection in this way *adds* complexity, by making a developer follow through definitions a few times before getting to the actual JSX which is rendered. 

But it's really not a big problem, just something I think we should be aware of, lest we end up which too many layers to be manageable :o